kaldb/src/main/java/com/slack/kaldb/metadata/core/KaldbMetadataStore.java

```diff
@@ -98,6 +99,22 @@ public T getSync(String path) {
     }
   }
 
+  public CompletionStage<Stat> hasAsync(String path) {
+    if (cachedModeledFramework != null) {
+      return cachedModeledFramework.withPath(zPath.resolved(path)).checkExists();
```

Comment:
is there an advantage of using the cacheModelFramework if it's present? i.e can `hasAsync` just use modeledClient?

Reply:
The benefits here would be consistency (ie, you would not have an issue where `has` returns a different result than `get` or `list`), and performance (can just check the cache).